google-cloud-spanner/test/google/cloud/spanner/snapshot/read_test.rb

```diff
@@ -225,6 +225,59 @@
     assert_results results
   end
 
+  it "can execute a simple read with directed read options" do
```

Comment:
I see all the test files have unit tests. Where are the integration tests which test a happy path, and the scenario where RW transaction returns error when set at request level?
Sorry if it is covered, but I am new to ruby and don't know how to find these.

Reply:
Happy path acceptance test is [here](https://github.com/googleapis/ruby-spanner/pull/75/files#diff-e07858e55ae77fd5ca0da96a9cd5f85d1e84b6f80b403320a7c610a5c3ccc4bd). There is no option exposed for users to set it in read_write transaction hence that may not be applicable here.

Reply:
Got it. Thanks.

Just noting down few points here for future reference. Ruby client is different from the golden client in the following,
1. In Ruby, each transaction type has its own `execute_query` and `read` methods exposed to customers. Example: Snapshot - [[execute_query](https://github.com/googleapis/ruby-spanner/blob/2659cac84d239c118f018304370b990179d1021e/google-cloud-spanner/lib/google/cloud/spanner/snapshot.rb#L304),  [read](https://github.com/googleapis/ruby-spanner/blob/2659cac84d239c118f018304370b990179d1021e/google-cloud-spanner/lib/google/cloud/spanner/snapshot.rb#L380)], RW Transaction [[execute_query](https://github.com/googleapis/ruby-spanner/blob/2659cac84d239c118f018304370b990179d1021e/google-cloud-spanner/lib/google/cloud/spanner/transaction.rb#L358), [read](https://github.com/googleapis/ruby-spanner/blob/2659cac84d239c118f018304370b990179d1021e/google-cloud-spanner/lib/google/cloud/spanner/transaction.rb#L711)]. That is why RW transaction does not expose any option to customers for setting dro. Whereas in golden clients, all transaction types refer to a common `execute_query` and `read` methods.